File: guet/commands/get.py
from guet.commands.command import Command
from guet.config.get_current_committers import get_current_committers
from guet.config.committer import Committer


class GetCommand(Command):
    def execute_hook(self):
        if self.args[0] == 'current':
            _print_current_committers()
        else:
            _print_invalid_identifier_message(self.args[0])

    @classmethod
    def help_short(cls):
        return ''

    def help(self):
        return """usage: guet get <identifier>\n\nValid Identifier\n\n\tcurrent - gets the currently set committers"""


def _print_current_committers():
    current = get_current_committers()
    print('Currently set committers')
    for committer in current:
        print(_format_committer(committer))


def _print_invalid_identifier_message(identifier: str):
    print(f'Invalid identifier <{identifier}>')


def _format_committer(committer: Committer):
    return f'{committer.initials} - {committer.name} <{committer.email}>'

File: src/commands/get.ts
import { ClosureChainLink, Command } from '../command';
import {
    Committer,
    getCurrentCommitters,
    getGlobalCommitters,
    getLocalCommitters,
} from '../committer';
import { log } from '../native-wrapper';

function getCommitters(args: string[]) {
    const [which] = args;
    switch (which) {
        case 'current':
            return currentCommitters();
        case 'all':
            return allCommitters();
    }
}

function currentCommitters() {
    log('Current committers:');
    getCurrentCommitters()
        .map(mapCommitter)
        .forEach(line => console.log(line));
}

function allCommitters() {
    log('All committers:');
    const globalCommitters = getGlobalCommitters();
    const localCommitters = getLocalCommitters();

    globalCommitters
        .map(committer => {
            if (
                localCommitters.find(
                    localCommiter =>
                        localCommiter.initials === committer.initials
                )
            ) {
                return `${mapCommitter(committer)} (overriden)`;
            } else {
                return mapCommitter(committer);
            }
        })
        .forEach(line => console.log(line));

    if (localCommitters.length > 0) {
        log('\n(local)');
        localCommitters.map(mapCommitter).forEach(line => console.log(line));
    }
}

function mapCommitter({ email, fullName, initials }: Committer) {
    return `${initials} - ${fullName} <${email}>`;
}

export const getCommand = new Command(
    'get',
    {
        description: 'Get information about the available committers.',
        usage: `guet get <identifier>
    current - lists currently set comitters
    all - lists all committers`,
    },
    new ClosureChainLink(getCommitters)
);
